front/lib/api/assistant/configuration.ts

```diff
@@ -247,6 +295,55 @@ export async function getAgentConfigurations(
   return [...globalAgents, ...agents];
 }
 
+async function filterByOverrides(
+  auth: Authenticator,
+  rawAgents: AgentConfiguration[]
+) {
+  const agentRelationOverrides = await getAgentRelationOverridesForUser(auth);
+  if (!agentRelationOverrides.isOk()) {
+    throw agentRelationOverrides.error;
+  }
+  const agentRelationOverrideMap = agentRelationOverrides.value;
+  return rawAgents.filter((a) => {
+    const override = agentRelationOverrideMap[a.sId];
+    if (override === "in-list" && a.scope === "published") {
+      return true;
+    } else if (override === "not-in-list" && a.scope === "workspace") {
+      return false;
+    } else {
+      return a.scope === "workspace";
+    }
+  });
+}
+
+async function getMentionsWithAgentId(
+  conversationId: string
+): Promise<Mention[]> {
```

Comment:
It's not OK to return a db type let's use a MentionType here and simplify the query?